app/code/Magento/Wishlist/Model/WishlistManagement.php

```diff
@@ -0,0 +1,227 @@
+
+            $wishlistData = [];
+            foreach ($collection as $item) {
+                $productInfo = $item->getProduct()->toArray();
+                $data = [
+                    "wishlist_item_id" => $item->getWishlistItemId(),
+                    "wishlist_id"      => $item->getWishlistId(),
+                    "product_id"       => $item->getProductId(),
+                    "store_id"         => $item->getStoreId(),
+                    "added_at"         => $item->getAddedAt(),
+                    "description"      => $item->getDescription(),
+                    "qty"              => round($item->getQty()),
+                    "product"          => $productInfo
+                ];
+                $wishlistData[] = $data;
+            }
+            return $wishlistData;
+        }
+    }
+
+    /**
+     * Add wishlist item for the customer
+     * @param int $customerId
+     * @param int $productIdId
+     * @return array|bool
+     *
+     */
+    public function addWishlistForCustomer($customerId, $productId)
+    {
+        if ($productId == null) {
+            $message = __('Invalid product, Please select a valid product');
+            $status = false;
+            $response[] = [
+                "message" => $message,
+                "status"  => $status
+            ];
+            return $response;
+        }
+        try {
+            $product = $this->_productRepository->getById($productId);
+        } catch (Exception $e) {
+            return false;
+        }
+        try {
+            $wishlist = $this->_wishlistFactory->create()
+                ->loadByCustomerId($customerId, true);
+            $wishlist->addNewItem($product);
+            $wishlist->save();
+        } catch (Exception $e) {
+            return false;
+        }
+        $message = __('Item added to wishlist.');
+        $status = true;
+        $response[] = [
+            "message" => $message,
+            "status"  => $status
+        ];
+        return $response;
+    }
+
+    /**
+     * Delete wishlist item for customer
+     * @param int $customerId
+     * @param int $productIdId
+     * @return array
+     *
+     */
+    public function deleteWishlistForCustomer($customerId, $wishlistItemId)
+    {
+
+        $message = null;
+        $status = null;
+        if ($wishlistItemId == null) {
+            $message = __('Invalid wishlist item, Please select a valid item');
+            $status = false;
+            $response[] = [
+                "message" => $message,
+                "status"  => $status
+            ];
+            return $response;
+        }
+        $item = $this->_itemFactory->create()->load($wishlistItemId);
+        if (!$item->getId()) {
+            $message = __('The requested Wish List Item doesn\'t exist .');
+            $status = false;
+
+            $response[] = [
+                "message" => $message,
+                "status"  => $status
+            ];
+            return $response;
```

Comment:
responses preparation in your case is a cross-cutting logic, which takes more code in this method than business logic, try to avoid  overload method with a lot of cross-cutting logic, because it's hard to read business logic and understand what method actually responsible for